src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -133,16 +131,13 @@ export class Kernel implements IKernel {
             return;
         } else {
             await this.validate(this.uri);
-            const metadata = ((getDefaultNotebookContent().metadata || {}) as unknown) as nbformat.INotebookMetadata;
-            // Create a dummy notebook metadata & update the metadata before starting the notebook (required to ensure we fetch & start the right kernel).
-            // Lower layers of code below getOrCreateNotebook searches for kernels again using the metadata.
-            updateNotebookMetadata(metadata, this.metadata);
             this._notebookPromise = this.notebookProvider.getOrCreateNotebook({
                 identity: this.uri,
                 resource: this.uri,
                 disableUI: options?.disableUI,
                 getOnly: false,
-                metadata,
+                metadata: undefined, // No need to pass this, as we have a kernel connection (metadata is required in lower layers to determine the kernel connection).
```

Comment:
Hmm, not sure on this. But due to the name changes and the difference between INotebookMetadata and KernelConnectionMetadata we have some stuff like this:
```
metadata: undefined,
kernelConnection: this.metadata
```
Which just scans odd to me. Looks funny to not have metadata be metadata (though I get why it's not). If we have more instances like this we could rename this.metadata in this class to kernelConnection to line the names up.

Reply:
Agreed, will fix.

Reply:
Fixed